src/main/kotlin/org/pkl/intellij/type/InferExprTypeFromContext.kt

```diff
@@ -283,6 +284,7 @@ private fun PklExpr?.doInferExprTypeFromContext(
             base.numberType -> base.numberType
             base.durationType -> base.durationType
             base.dataSizeType -> base.dataSizeType
+            base.bytesType -> base.bytesType ?: base.additiveOperandType
```

Comment:
This confused me a bit, but this is correct. Maybe let's add a comment?

```suggestion
            base.bytesType -> base.bytesType
              // if we fall through, both bytes type and unaliased type is [null] (possible when Pkl < 0.29)
              ?: base.additiveOperandType
```